src/h3/mod.rs

```diff
@@ -1311,11 +1293,29 @@ impl Connection {
                     }
                 },
 
-                stream::State::QpackInstruction => return Err(Error::Done),
+                stream::State::QpackInstruction => {
+                    let e = match stream.ty() {
+                        Some(stream::Type::QpackEncoder) =>
+                            Error::QpackEncoderStreamError,
+                        Some(stream::Type::QpackDecoder) =>
+                            Error::QpackDecoderStreamError,
+
+                        _ => unreachable!(),
+                    };
+
+                    conn.close(true, e.to_wire(), b"")?;
+
+                    return Err(e);
+                },
 
-                stream::State::Done => return Err(Error::Done),
+                stream::State::Drain => {
+                    let mut d = [0; 4096];
```

Comment:
Now that the Drain states seems to handle things a little differently, one thing we could do is send a STOP_SENDING. This could prevent us from wasting cycles on handling streams that are useless.

Reply:
Or, am I missing how Drain works in practice...

Reply:
This is pretty much equivalent to the previous behavior, with the difference that instead of filling the stream buffer, we just discard the data immediately.

But yes, we could probably send STOP_SENDING, if we supported it.

Reply:
Something else we could do, now that reading is more dynamic, is not drain unknown stream types. That gives us some way to put back pressure on a stream.

Reply:
So, that is how I implemented this originally (basically just return `Error::Done` like it was done before), but for whatever reason it broke fetching big files from cloudflare-quic.com. That is, the http3-client wouldn't receive all of the data anymore, and then the connection would time out without receiving the `Finished` event.

I just tried again (just reaplaced this block with `reuturn Err(Error::Done);` and I can still reproduce the problem, though it might just be that some changes in some other parts of the code are also needed to make this work.

We might want to do some more investigation in the future when we have more time.

Reply:
Agreed. This sounds like a separate optimisation task to investigate STOP_SENDING and lazy reads. No need to block this PR on that.